### ISSUE TITLE
A question about custom loss function

### ISSUE THREAD

**Issue by yscacaca** [fchollet/keras#3058]
Hi,

I am using a custom loss function with my keras model. In my custom loss function, it returns "an array" _v_ with shape (num_of_sample, ) instead of a "scalar" likes those in [https://github.com/fchollet/keras/blob/master/keras/objectives.py](url). 

But the model with this custom loss function can still be compiled and trained. I just wondering what happens in Keras when I return "an array" _v_ in my custom loss function. Will it automatically adds code like _K.sum(v)_ , _K.mean(v)_ or _v[0]_ ?

Thanks
